opm/simulators/wells/BlackoilWellModelGeneric.cpp

```diff
@@ -987,24 +987,25 @@ updateWsolvent(const Group& group,
         if (std::abs(gasProductionRate) > 1e-6)
             wsolvent = solventProductionRate / gasProductionRate;
 
-        setWsolvent(group, reportStepIdx, wsolvent);
+        setWsolvent(group, reportStepIdx, wellState, wsolvent);
     }
 }
 
 template<class Scalar>
 void BlackoilWellModelGeneric<Scalar>::
 setWsolvent(const Group& group,
             const int reportStepIdx,
+            const WellState<Scalar>& wellState,
             Scalar wsolvent)
 {
     for (const std::string& groupName : group.groups()) {
         const Group& groupTmp = schedule_.getGroup(groupName, reportStepIdx);
-        setWsolvent(groupTmp, reportStepIdx, wsolvent);
+        setWsolvent(groupTmp, reportStepIdx, wellState, wsolvent);
     }
 
     for (const std::string& wellName : group.wells()) {
         const auto& wellTmp = schedule_.getWell(wellName, reportStepIdx);
-        if (wellTmp.getStatus() == Well::Status::SHUT)
+        if (!wellState.has(wellName) || wellState.well(wellName).status == WellStatus::SHUT)
```

Comment:
i think it should be enough to check if wellName is in generic_wells_

Reply:
yeah, you're right, I'm just too lazy to look up values in vectors.. Maybe well_container_generic_ should be a WellContainer?

Reply:
thanks, will do

Reply:
https://github.com/OPM/opm-simulators/pull/5849 got around to it, so this will be a simple 
```c++
if (!hasOpenWell(foo)) {
    continue;
}
``` 

which makes intent very readable.